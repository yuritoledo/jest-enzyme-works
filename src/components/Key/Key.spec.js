import React from 'react'
import { shallow } from 'enzyme'
import Key from './Key'

describe('Key Component', () => {
	let wrapper = shallow(<Key keyAction={jest.fn()} keyType={''} keyValue={''} />)
	beforeEach(() => {
		wrapper = shallow(
			<Key
				keyAction={jest.fn()}
				keyType={''}
				keyValue={''}
			/>
		)
	})

	test('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('should render a <div />', () => {
		expect(wrapper.find('div')).toHaveLength(1)
	})

	test('should render the value of keyvalue	', () => {
		wrapper.setProps({ keyValue: 'test' })
		expect(wrapper.text()).toEqual('test')
	})




})