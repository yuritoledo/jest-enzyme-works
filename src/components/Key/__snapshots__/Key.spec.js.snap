// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Key
    keyAction={[Function]}
    keyType=""
    keyValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="key-value"
/>,
      "className": "key-container ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "key-value",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="key-value"
/>,
        "className": "key-container ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "key-value",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
